### PR TITLE
Add additional log details for database errors

### PR DESCRIPTION
**Pull request by mprahl**
Relates:
https://issues.redhat.com/browse/ACM-10180

### CODE REVIEW

File: controllers/complianceeventsapi/server.go [hunk 1]
```diff
@@ -754,6 +754,26 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter,
...
+
+	for unwrappedErr != nil {
+		if pqErr, ok := unwrappedErr.(*pq.Error); ok { //nolint: errorlint
+			return append(
+				[]interface{}{"dbMessage", pqErr.Message, "dbDetail", pqErr.Detail, "dbCode", pqErr.Code},
+				additionalKeyVals...,
+			)
+		}
+
+		unwrappedErr = errors.Unwrap(unwrappedErr)
```
Comment (JustinKuli) [inline controllers/complianceeventsapi/server.go:771]: This seems like the kind of thing that errors.Is or errors.As would do... but I readily admit being confused by those often. Do neither of those work here?

File: controllers/complianceeventsapi/server.go [hunk 2]
```diff
@@ -754,6 +754,26 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter,
 	}
 }
 
+// getPqErrKeyVals is a helper to add additional database error details to a log message. additionalKeyVals is provided
+// as a convenience so that the keys don't need to be explicitly set to interface{} types when using the
+// `getPqErrKeyVals(err, "key1", "val1")...“ syntax.
+func getPqErrKeyVals(err error, additionalKeyVals ...interface{}) []interface{} {
+	unwrappedErr := err
+
+	for unwrappedErr != nil {
```
Comment (yiraeChristineKim) [inline controllers/complianceeventsapi/server.go:763]: Why this is `for`?

Comment (mprahl) [reply controllers/complianceeventsapi/server.go:763]: The idea is to keep unwrapping until either a pq.Error is found or we've unwrapped all that we could and thefore, nil is returned.

Comment (yiraeChristineKim) [reply controllers/complianceeventsapi/server.go:763]: oh I see so unwrap one by one then log db errors each

**Review by yiraeChristineKim — APPROVED** [review]

File: controllers/complianceeventsapi/server.go (hunk 1 above)
Comment (JustinKuli) [reply controllers/complianceeventsapi/server.go:771]: I think `errors.As` could work as a type conversion: [errors.As](https://pkg.go.dev/errors) "examines the tree of its first argument looking for an error that can be assigned to its second argument, which must be a pointer. If it succeeds, it performs the assignment and returns true."

But I mention it more out of curiosity than necessity.

Comment (mprahl) [reply controllers/complianceeventsapi/server.go:771]: Oh yeah, you're right. I didn't catch that the second argument is a pointer to the type that gets set if the type matches:
https://github.com/open-cluster-management-io/governance-policy-propagator/pull/202

**Pull request merged**